lib/ansible/template/__init__.py

```diff
@@ -412,9 +412,30 @@ def __init__(self, delegatee, pluginloader, *args, **kwargs):
 
         self._collection_jinja_func_cache = {}
 
+        self._ansible_plugins_loaded = False
+
+    def _load_ansible_plugins(self):
+        if self._ansible_plugins_loaded:
+            return
```

Comment:
Will roles that provide filters or tests, and are loaded using `include_role` end up being excluded due to this?

Reply:
Not according to my testing. That check is basically an equivalent of what we currently do:
https://github.com/ansible/ansible/blob/9792d631b1b92356d41e9a792de4b2479a1bce44/lib/ansible/template/__init__.py#L628

From my understanding it works (this PR and the current code) also for roles due to us creating templars several times so filters/tests are loaded correctly for roles.

We might want to make some adjustments/optimizations but for this PR I just wanted to move `get_filters/get_tests`.

We have an integration test that somewhat tests this https://github.com/ansible/ansible/blob/devel/test/integration/targets/template/filter_plugins.yml.

Reply:
+1